src/imgui/ImGuiSymbols.cc

```diff
@@ -171,10 +238,23 @@ static void drawTable(ImGuiManager& manager, const SymbolManager& symbolManager,
 	});
 }
 
-void ImGuiSymbols::paint(MSXMotherBoard* /*motherBoard*/)
+static int toSlotIndex(MSXMotherBoard* motherBoard, std::optional<int> slot, std::optional<int> subslot)
+{
+	if (!slot) return 0;
+
+	auto& cpuInterface = motherBoard->getCPUInterface();
+	int slotIndex = 1;
+	for (int ps = 0; ps < slot; ++ps) {
+		slotIndex += cpuInterface.isExpanded(ps) ? 4 : 1;
+	}
+	return slotIndex + (subslot ? *subslot : 0);
+}
```

Comment:
When we remove the `index` field from `FileInfo` this helper function isn't needed anymore. More details below.

Note that `motherBoard` can be `nullptr` (when there's no active MSX machine). In that case this function would have crashed.  (When there's no active machine you can e.g. assume that all primary slots are expanded).

Reply:
Code is getting a bit hard to maintain. Maybe `Reactor::getMotherBoard()` should implement the null object pattern and replace `motherBoard` with a dummy object instead of returning a `nullptr`.